sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -360,6 +367,75 @@ def retry_setup_grpc_channel_for_port(port, asic_index):
                 grpc_port_stubs[port] = stub
                 return True
 
+
+def get_grpc_credentials(type, kvp):
+
+    root_file = kvp.get("ca_crt", None)
+    if root_file is not None: 
+        root_cert = open(root_file, 'rb').read()
+    else:
+        helper_logger.log_error("grpc credential channel setup no root file in config_db")
+        return None
+
+    if type == "mutual":
+        cert_file = kvp.get("client_crt", None)
+        if cert_file is not None: 
+            cert_chain = open(cert_file, 'rb').read()
+        else:
+            helper_logger.log_error("grpc credential channel setup no cert file for mutual authentication in config_db")
+            return None
+
+        key_file = kvp.get("client_key", None)
+        if key_file is not None: 
+            key = open(key_file, 'rb').read()
+        else:
+            helper_logger.log_error("grpc credential channel setup no key file for mutual authentication in config_db")
+            return None
+
+        credential = grpc.ssl_channel_credentials(
+                root_certificates=root_cert,
+                private_key=key,
+                certificate_chain=cert_chain)
+    elif type == "server":
+        credential = grpc.ssl_channel_credentials(
+                root_certificates=root_cert)
+    else:
+        #should not happen
+        helper_logger.log_error("grpc credential channel setup no type specified for authentication in config_db")
+        return None
+
+    return credential
+
+def create_channel(type,level, kvp, soc_ip):
+
+    retries = 3
+    for _ in range(retries):
+
+        if type == "secure": 
+            credential = get_grpc_credentials(type, kvp)
```

Comment:
this should be
```suggestion
            credential = get_grpc_credentials(level, kvp)
```
Right?

Reply:
Thanks for pointing out, fixed.